treestats_paper.tex

```diff
@@ -252,10 +265,12 @@ \section*{Framework and statistics}
 i.e., $\iw(v)$ may be a vector $(\iw_1(v), \ldots, \iw_m(v))$,
 so that when summarising we have access to more than one aspect of each subtree.
 The requirement that $f(0) = f(\tiw) = 0$ is for consistency,
-so that statistics do not depend on portions of the tree not ancestral to any of the samples.
+so that statistics do not depend on portions of the tree either not ancestral to any of the samples or ancestral to all
+of them.
```

Comment:
since we describe also polarised statistics below, that *can* depend on fixed mutations, maybe we shouldn't add this extra caveat?

Reply:
But, they (polarised states) can't, as our email exchange has shown.

Reply:
They *can*, with the argument `strict=False`. Unless I'm missing something?

Reply:
I think this is where we disagree on design.  If there is to be one function with `mode` defining the behavior, then all input functions, `f`, should be implementable with the same rules.  Right now, that is not true:

* If `strict == True`, a function must skip fixations.  However, that is not possible in Python w/o a more complex design for `f` in general, as the subsample sizes are not passed to `f`.
* If `strict == False`, anything goes.
* If `polarised` is `False`, we want the output of `f` to be invariant to monomorphic positions.
* If `polarised` is `True`, we don't care.  (I have to look back to see if that is clear on the docs and in the manuscript--why don't we care here?  For example, we don't want Fay and Wu's H to depend on, say, the number of neutral fixations, which are a function of how long a simulation has been run.)

So, that is a 2x2 matrix of possibilities affecting how `f`is correctly implemented.

Reply:
Hm, that's not how I'd summarise it. Here's my summary:

1. In the paper, we define abstractly "polarisation" for a statistic, which determines whether the ancestral allele is included in the sum or not. Was this clear in the current write-up?
2. In the code, the `general_stat` and `sample_count_stat` interfaces include a `polarised` option that implements this dichotomy.
3. Statistics that only depend on the genotype matrix without external information about which alleles are ancestral should ignore monomorphic sites, so to aid in debugging, we put in a check for this property that can be turned off with the `strict=False` option.

> a function must skip fixations. However, that is not possible in Python w/o a more complex design for f in general, as the subsample sizes are not passed to f.

I don't agree that the design is more complex, but this seems to be a matter of aesthetics, so we should not argue about it.

Reply:
>     1. In the paper, we define abstractly "polarisation" for a statistic, which determines whether the ancestral allele is included in the sum or not. Was this clear in the current write-up?

Inclusion of the ancestral allele vs not doesn't imply anything about whether a fixation is allowed.  To make that clear, we have to say that the presence of the ancestral allele is not required, or something along those lines.

To me, it is a partially a matter of aesthetics, but also a matter of why the default isn't `strict=False`, which seems more general and in the spirit of things.

Reply:
> Inclusion of the ancestral allele vs not doesn't imply anything about whether a fixation is allowed.

Right, I agree. The `strict` or not thing is orthogonal to polarisation, as you also said.

>  To make that clear, we have to say that the presence of the ancestral allele is not required, or something along those lines.
> ...
> To me, it is a partially a matter of aesthetics, but also a matter of why the default isn't strict=False, which seems more general and in the spirit of things.

Ok, how about we add a paragraph about the motivation behind `strict`, and other considerations (as started [here](https://github.com/tskit-dev/tskit/issues/357)). This should explain why/when you'd want to restrict to `strict` functions.  I can write this, and would vote to put this before the "Duality..." section. How's that sound - I can draft this and we can discuss more at that point?

And, fwiw, the reason that `strict=True` is the default is that, IMO, usually if someone has defined a function that would fail the `strict` check, then they've made an error (since they'd get a stat that depends on the monomorphic sites, etc etc).

Reply:
Sounds good.  Perhaps we should discuss there, but I am confused about your last statement vs a previous one.  Once `polarised=True`, an `f` failing the strict check is no longer to be considered an error.